File: examples/python/smoke.py
import importlib.util
import sys


def load_addon(path):
    spec = importlib.util.spec_from_file_location("dynabridge_python_example", path)
    if spec is None or spec.loader is None:
        raise RuntimeError(f"cannot load extension module from {path}")
    module = importlib.util.module_from_spec(spec)
    spec.loader.exec_module(module)
    return module


if len(sys.argv) != 2:
    raise RuntimeError("usage: smoke.py <addon>")

addon = load_addon(sys.argv[1])
assert addon.add(12, 13) == 25

counter = addon.counter(13)
assert counter.add(29) == 42
assert counter.value() == 13

File: examples/napi/smoke.js
const assert = require("assert");

const addonPath = process.argv[2];
assert.ok(addonPath, "expected addon path");

const addon = require(addonPath);
assert.strictEqual(addon.add(12, 13), 25);

const counter = new addon.counter(13);
assert.strictEqual(counter.add(29), 42);
assert.strictEqual(counter.value(), 13);
